inc/azure_c_util/interlocked.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef INTERLOCKED
+#define INTERLOCKED
+
+#ifdef __cplusplus
+#include <cstdint>
+#include <cstdbool>
```

Comment:
This isn't needed because it's an intrinsic type in c++, you can remove this line